Shimakaze-chan/Utils.cs

```diff
@@ -38,6 +39,39 @@ public static bool MemberHasPermissions(DiscordMember member, Permissions permis
             return false;
         }
 
+        /// <summary>
+        /// This gets BOTH the mentioned users and parses the message for ulong Ids such as UIDs
+        /// </summary>
+        /// <param name="mentionedUsers">ctx.Message.MentionedUsers</param>
+        /// <param name="commandPayload">message string from [RemainingText]</param>
```

Comment:
yes but actually no. the description is actually wrong.
it shouldn't be expected to work with the [RemainingText] string itself. (i.e. it could contain some other data like how many message to delete of those users)
```suggestion
        /// <param name="commandPayload">string containing the list of user IDs separated by spaces</param>
```

and the name would be ambiguous? (not representative of what it actually is), so change that too if you're changing the description

Reply:
The command will not error out regardless of [Remaining Text] content though, so even if it's not expected it will work.
And it has to be expected to work with solely remaining text itself cause that's how You wanted to provide Ids. Pings in this string are redundant cause they are covered by MentionedUsers list.

What is very confusing is it requires [Remaining Text] even if you don't parse it at all and just take the MentionedUsers list...

But yes, the description should be changed cause it can work with strings other than [Remaining Text] too. The name i dunno... messageString would be wrong as well, stringToParse or stringWithUIds sounds pretty poor though.

Reply:
> The command will not error out regardless of [Remaining Text] content though, so even if it's not expected it will work.

the function will not error out, no. but the command will, again lets take the example of deleting messages:
usage: purge <.count> <[ids]>
ex: purge 10 123444444 9888888888
you drop full RemainingText in the function, it will return an array of size 3: [10, 1234..., 98....]
the command that called the function will now expect the first item to be a user id: "user id 10 not found in guild". failed.
> And it has to be expected to work with solely remaining text itself cause that's how You wanted to provide Ids

explained how it shouldn't be, and me wanting to provide IDs does not mean that that is the only thing that i want the RemainingText content to be, as proven above


and uh.. idk UserIDsString ?

Reply:
Oh I see what you meant, yes, I should not suggest it's pattern matching to only ever extract Id ulongs.

Or change the Util to pattern match the ulongs by word length... It is a simple regex so if we want to reuse the Util for mixed [Remaining Text] content i guess it's possible, but then it's gonna be a problem to distinguish user ulong from channel ulong etc.

Reply:
it's easier to know the context of what you want and just pass it what you should pass it tbh

and wouldn't really make sense to use that function for channels/etc. because of the mentioned users as well.